python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -918,3 +942,218 @@ def test_validate_load_balancer_managed_outbound_ip_count(test_description, name
+        None
+    ),
+    (
+        "Should raise RequiredArgumentMissingError if enable_managed_identity is not present",
+        Mock(enable_managed_identity=None,
+             platform_workload_identities=[]),
+        RequiredArgumentMissingError,
+        None
+    ),
+    (
+        "Should raise RequiredArgumentMissingError if enable_managed_identity is False",
+        Mock(enable_managed_identity=False,
+             platform_workload_identities=[]),
+        RequiredArgumentMissingError,
+        None
+    ),
+    (
+        "Should raise InvalidArgumentError if any resource IDs are not for userAssignedIdentities",
+        Mock(enable_managed_identity=True,
+             subscription_id="00000000-0000-0000-0000-000000000000",
+             resource_group_name="resourceGroup",
+             platform_workload_identities=[
+                 Mock(resource_id="/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup/providers/Microsoft.Network/virtualNetworks/foo"),
+             ]),
+        InvalidArgumentValueError,
+        None
+    ),
+    (
+        "Should convert all identities passed in as names to full resource IDs",
+        Mock(enable_managed_identity=True,
+             subscription_id="00000000-0000-0000-0000-000000000000",
+             resource_group_name="resourceGroup",
+             platform_workload_identities=[
+                 Mock(resource_id="/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/anotherResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/foo"),
+                 Mock(resource_id="bar")
+             ]),
+        None,
+        [
+            Mock(resource_id="/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/anotherResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/foo"),
+            Mock(resource_id="/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/bar"),
+        ]
+    )
+]
+
+
+@pytest.mark.parametrize(
+    "test_description, namespace, expected_exception, expected_identities",
+    test_validate_platform_workload_identities_data,
+    ids=[i[0] for i in test_validate_platform_workload_identities_data]
+)
+def test_validate_platform_workload_identities(test_description, namespace, expected_exception, expected_identities):
+    cli_ctx = Mock(data={'subscription_id': namespace.subscription_id})
+    cmd = Mock(cli_ctx=cli_ctx)
+    if expected_exception is None:
+        validate_platform_workload_identities(cmd, namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_platform_workload_identities(cmd, namespace)
+
+    if expected_identities is not None:
+        for expected, actual in zip(expected_identities, namespace.platform_workload_identities):
+            assert (expected.resource_id == actual.resource_id)
+
+
+test_validate_cluster_identity_data = [
```

Comment:
This list could use a `Should not raise any exception when valid` case where the identity is passed as a resource ID.